pkg/component/kubernetes/apiserverexposure/service.go

```diff
@@ -126,10 +126,6 @@ func (s *service) Deploy(ctx context.Context) error {
 	obj := s.emptyService()
 
 	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, s.client, obj, func() error {
-		// TODO(scheererj): Remove this after release of v1.125
-		// Clean up old annotations as unnecessary load balancer annotations may reside on a ClusterIP service.
-		obj.Annotations = map[string]string{}
```

Comment:
Can we also adapt the unit test to cover that an existing annotation won't be removed?

You could bring back the change in the test from https://github.com/gardener/gardener/pull/12630/files.